boilerplates/lucia-auth/files/libs/db.ts

```diff
@@ -0,0 +1,30 @@
+import sqlite, { type Database } from "better-sqlite3";
+
+export const db: Database = sqlite(":memory:");
+
+db.exec(`CREATE TABLE IF NOT EXISTS users (
+    id TEXT NOT NULL PRIMARY KEY,
+    username TEXT NOT NULL,
+    password_hash TEXT
+)`);
+
+db.exec(`CREATE TABLE IF NOT EXISTS oauth_accounts (
+    provider_id TEXT NOT NULL,
+    provider_user_id TEXT NOT NULL,
+    user_id TEXT NOT NULL,
+    PRIMARY KEY (provider_id, provider_user_id),
+    FOREIGN KEY (user_id) REFERENCES users(id)
+)`);
+
+db.exec(`CREATE TABLE IF NOT EXISTS sessions (
+    id TEXT NOT NULL PRIMARY KEY,
+    expires_at INTEGER NOT NULL,
+    user_id TEXT NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES users(id)
+)`);
+
+export interface DatabaseUser {
+  id: string;
+  username: string;
+  password_hash?: string;
+}
```

Comment:
Adds links to the lucia-auth docs where we can (like https://lucia-auth.com/database/sqlite)